### PR TITLE
BP/551/develop ad system

### PR DESCRIPTION
**Pull request by bhupeshappfoster**
### Description:
[Provide a brief description of the purpose and goal of this pull request.]

### Related Issue(s):

- [#551](https://github.com/AcclaroInc/pm-craft-translations/issues/551)

### Changes Made:
[List the changes made in this pull request to address the issue or implement the feature.]

**Added:**

- 

**Changed:**

-

**Deprecated:**

-

**Removed:**

-

**Fixed:**

-

**Security:**

-

### Testing:
[Describe the testing performed to ensure the changes are functioning as expected.]

- 

### Quality Assurance (QA):

- [ ] The code has been reviewed and approved by the QA team.
- [ ] The changes have been tested on different environments (e.g., staging, production).
- [ ] Integration tests have been performed to verify the interactions between components.
- [ ] Performance tests have been conducted to ensure the changes do not impact system performance.
- [ ] Any necessary database migrations or data transformations have been executed successfully.
- [ ] Accessibility requirements have been considered and tested (e.g., screen reader compatibility, keyboard navigation).

### Resources:

**Screenshots (if applicable):**
In dashboard
image
In Existing order/translator sidebar
image
In create order/translator sidebar
image

### Wrapping up

**Checklist:**

- [ ] The code builds without any errors or warnings.
- [ ] The code follows the project's coding conventions and style guidelines.
- [ ] Unit tests have been added or updated to cover the changes made.
- [ ] The documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing tests pass successfully.
- [ ] The PR has been reviewed by at least one other team member.

**Additional Notes:**
[Include any additional notes, considerations, or context that may be relevant.]

### CODE REVIEW

**Comment by bhupeshappfoster** [comment]
@sharmarishabh1 Changes deployed on dev server for testing.

CC: @MerrittBoydston  @sidedwards

**Review by sharmarishabh1 — COMMENTED** [review]
Just one small review. 

Also, can you please look at this screenshot here?

image

@bhupeshappfoster

File: src/services/repository/AdsRepository.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	public function sidebar($context)
+	{
+		return $this->getAdsContent("sidebar", $context);
+	}
+
+	public function footer()
+	{
+		return $this->getAdsContent("footer");
+	}
+
```
[the remark this answers is not in the document]
Comment (bhupeshappfoster) [reply src/services/repository/AdsRepository.php]: resolved

**Comment by bhupeshappfoster** [comment]
> Just one small review.
> 
> Also, can you please look at this screenshot here?
> 
> image
> 
> @bhupeshappfoster

Fixed

**Comment by sidedwards** [comment]
@bhupeshappfoster the Ad cta Burton should be left aligned

File: src/Constants.php [hunk 2]
```diff
@@ -324,10 +324,10 @@ class Constants
...
     ];
 
-    const CTA_PROFESSIONAL_TRANSLATION = "https://www.acclaro.com/get-started/translation-quote/";
-    const CTA_QUALITY_REVIEW = "https://www.acclaro.com/localization/";
-    const CTA_TECHNICAL_SUPPORT = "https://www.acclaro.com/localization-partner/";
-    const CTA_AI_ML_TRANSLATION = "https://www.acclaro.com/localization/";
+    const CTA_PROFESSIONAL_TRANSLATION = "https://www.acclaro.com/solutions/global-marketing/website-localization-services";
+    const CTA_QUALITY_REVIEW = "https://www.acclaro.com/solutions/language-quality";
+    const CTA_TECHNICAL_SUPPORT = "https://info.acclaro.com/craftcms-technical-support";
+    const CTA_AI_ML_TRANSLATION = "https://info.acclaro.com/translation-plugin-for-craft3-support-info";
```
Comment (sidedwards) [inline src/Constants.php:330]: @bhupeshappfoster @sharmarishabh1 are these constants also used for the services page?

Also, the value for `CTA_AI_ML_TRANSLATION` is incorrect. We can just have an empty string for now.

File: src/Constants.php [hunk 3]
```diff
@@ -379,4 +379,30 @@ class Constants
...
+
+    // Dashboard will be array of ads to show more ads if needed in future
+    const ADS_CONTENT = [
+        "dashboard" => [
+            [
+                "heading" => "AI/Machine Translation",
+                "content" => "Translate more content faster with AI.",
+                "linkText" => "Try it now",
+                "link" => self::CTA_AI_ML_TRANSLATION,
+            ]
```
Comment (sidedwards) [inline src/Constants.php:391]: We should not display an ad for AI/MT yet.

File: src/templates/_components/ads/sidebar.twig [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	<div class="ad-container sidebar-ad">
+		<div class="field">
+			<div class="heading">
+				<label> {{ heading }} </label>
+			</div>
+
+			<div class="input ltr">
+				<label> {{ content }} </label>
+			</div>
+			<div class="flex" style="justify-content: end;">
```
Comment (sidedwards) [inline src/templates/_components/ads/sidebar.twig:18]: As mentioned here: https://github.com/AcclaroInc/craft-translations/pull/527#issuecomment-2422857319

The button should be aligned left

File: src/templates/_components/widgets/AcclaroAds/body.twig [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                            label: ad.linkText,
+                            href: ad.link,
+                        }) }}
+                    </div>
+                </div>
+            {% endfor %}
+        </div>
+    {% else %}
+        <div class="data fullwidth">
+            <p>{{ "No ads to show."|t('app') }}</p>
```
Comment (sidedwards) [inline src/templates/_components/widgets/AcclaroAds/body.twig:25]: If there aren't any available, we should just not display anything.

**Review by sidedwards — CHANGES_REQUESTED** [review]
@bhupeshappfoster please see feedback.

File: src/Constants.php (hunk 2 above)
Comment (sharmarishabh1) [reply src/Constants.php:330]: @sidedwards  Yes, we are using the same on the services page too. 

@bhupeshappfoster 
Can you please add "#" here instead?

File: src/Constants.php (hunk 3 above)
Comment (bhupeshappfoster) [reply src/Constants.php:391]: Done

File: src/Constants.php (hunk 2 above)
Comment (bhupeshappfoster) [reply src/Constants.php:330]: Done

File: src/templates/_components/widgets/AcclaroAds/body.twig (hunk 5 above)
Comment (bhupeshappfoster) [reply src/templates/_components/widgets/AcclaroAds/body.twig:25]: done

File: src/templates/_components/ads/sidebar.twig (hunk 4 above)
Comment (bhupeshappfoster) [reply src/templates/_components/ads/sidebar.twig]: done

**Review by sidedwards — APPROVED** [review]

**Pull request merged**